MIDAS/src/hardware/GPSSensor.cpp

```diff
@@ -17,14 +17,18 @@ ErrorCode GPSSensor::init() {
     return ErrorCode::NoError;
 }
 
+// GPS Coordinates are ddmm.mmmm, so please convert with some code
 GPS GPSSensor::read() {
     teseo.update();
     GPGGA_Info_t gpgga_message = teseo.getGPGGAData();
     GPRMC_Info_t gprmc_message = teseo.getGPRMCData();
-    // Divide latitude and longitude by 100 because NMEA data is ddmmm.mmm or something like that
     // Also check if it's north or south
-    float lat = gpgga_message.xyz.lat / 100.f * ((gpgga_message.xyz.ns == 'N') ? 1 : -1);
-    float lon = gpgga_message.xyz.lon / 100.f * ((gpgga_message.xyz.ns == 'E') ? 1 : -1);
+    float lat = gpgga_message.xyz.lat * ((gpgga_message.xyz.ns == 'N') ? 1. : -1.);
+    float lon = gpgga_message.xyz.lon * ((gpgga_message.xyz.ns == 'E') ? 1. : -1.);
```

Comment:
These number should not be floats. Lat and lon need to be doubles to avoid rounding errors